### DISCUSSION TITLE
Correct SOQL syntax using AND or && in a where clause

### DISCUSSION THREAD

**Discussion by baobao917** [jamessimone/apex-rollup#510] [Q&A] [answered]
I know that the Child Object Where Clause field is finicky when it comes to the syntax.  I am attempting to introduce to criteria in my WHERE clause.  I've written it multiple ways and it just doesn't work.  Below are the different examples:

1. Order_Status__c = 'Order Shipped (Complete)' && Parent_WF_Order__c != NULL

2. Order_Status__c = 'Order Shipped (Complete)' && NOT(ISBLANK(Parent_WF_Order__c))

3. AND(Order_Status__c = 'Order Shipped (Complete)', NOT(ISBLANK(Parent_WF_Order__c)))

None of these work.  The 3rd one actually throws out an error.  The first two nothing happens.  When I simplify the WHERE statement to only one condition (Order_Status__c = 'Order Shipped (Complete)'), it works just fine.

Guidance?

**Comment by jamessimone** [comment, marked as answer]
The correct syntax is SOQL-compliant:
![image](https://github.com/jamessimone/apex-rollup/assets/16430727/65761a24-b4e0-440e-a394-cd3bf2779063)

This isn't valid SOQL (let alone a sensible query!), but if you wanted to do the same thing, you'd do `condition1 AND condition 2`.

In other words - the where clause doesn't work like a validation rule (where formulas and the usage of syntax like `AND(condition1, condition2)` is supported), it works like a SOQL where clause (with some limitations, and sometimes creating a formula field that evaluates to true or false would be easier, because I can't promise that I've built a fully compliant parser for where clauses). 

Of course, I'm trying as much as possible to support _fully_ SOQL compliant where clause parsing, so please let me know if there's something that's not working for you on that front.

**Reply by baobao917** [reply]
Thanks @jamessimone for the clarification!  Somehow in my head I got mixed up with SOQL syntax vs SF formula syntax.  Works now!

**Reply by jamessimone** [reply]
No worries whatsoever. I've definitely had my fair share of similar mixups. I've also found myself thinking _wouldn't it be fun to support formula syntax there?_ but in the end I think sticking with how the platform also behaves - as much as possible - is the way to go
